examples/advanced/advanced/src/screens/Camera.js

```diff
@@ -601,6 +645,35 @@ class Camera extends Component{
 
     this.cameraStyle = cameraStyle;
 
+    let whiteBalance;
+    let customWBView;
+    if (wb >= WB_OPTIONS.length) {
+      let currentOption = this.state.currentCustomWBOption;
+      let currentValue = this.state.customWhiteBalance[currentOption];
+      let options = CUSTOM_WB_OPTIONS_MAP[currentOption]
+      customWBView = (
+        <View style={styles.customWBView}>
+          <Button style={styles.customWBViewButton} onPress={this.changeCustomWBOption}>
+            <Text style={styles.customWBViewText}>{options.label}</Text>
+          </Button>
+          <Slider
+            style={styles.customWBViewSlider}
+            value={currentValue}
+            step={options.steps}
+            minimumValue={options.min}
+            maximumValue={options.max}
+            minimumTrackTintColor="#FFFFFF"
+            maximumTrackTintColor="#000000"
+            onValueChange={this.changeCustomWBOptionValue}
+          />
+          <Text style={[styles.customWBViewText, {minWidth: '15%'}]}>{currentValue.toFixed(1)}</Text>
+        </View>
+      );
+      whiteBalance = this.state.customWhiteBalance;
+    } else {
+      whiteBalance = WB_OPTIONS[wb];
+    }
```

Comment:
I know that's an example of code, but I think it's pretty much harder than it should be, so what do you think about the following suggestion?

```suggestion
    let whiteBalance = wb >= WB_OPTIONS.length ? this.state.customWhiteBalance : WB_OPTIONS[wb];
    const { currentCustomWBOption } = this.state;
    let customWhiteBalanceValue = this.state.customWhiteBalance[currentCustomWBOption];
    let customWhiteBalanceOption = CUSTOM_WB_OPTIONS_MAP[currentCustomWBOption]
```

Reply:
Yes, you are right. It's better to keep the UI code in one place and the logic in another.
I will also add this line of code `let isCustomWhiteBalance = wb >= WB_OPTIONS.length;` to have the custom white balance check only once.